plugins/modules/backup_plan.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import json
+from ansible.module_utils.resource_tags import ensure_tags
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+
+DOCUMENTATION = r"""
+module: backup_plan
+short_description: create, delete and modify AWS Backup plans
+version_added: 6.0.0
+description:
+  - Manages AWS Backup plans.
+  - For more information see the AWS documentation for backup plans U(https://docs.aws.amazon.com/aws-backup/latest/devguide/about-backup-plans.html).
+options:
+  backup_plan_name:
+    description:
+      - The display name of a backup plan. Must contain 1 to 50 alphanumeric or '-_.' characters.
+    required: true
+    type: str
+  rules:
+    description:
+      - An array of BackupRule objects, each of which specifies a scheduled task that is used to back up a selection of resources.
+    required: false
+    type: list
+  advanced_backup_settings:
+    description:
+      -  Specifies a list of BackupOptions for each resource type. These settings are only available for Windows Volume Shadow Copy Service (VSS) backup jobs.
+    required: false
+    type: list
+  state:
+    description:
+      - Create, delete a backup plan.
+    required: false
+    default: present
+    choices: ['present', 'absent']
+    type: str
+notes: []
+author:
+  - Kristof Imre Szabo (@krisek)
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+  - amazon.aws.tags
+"""
+EXAMPLES = """
+- name: create backup plan
+  backup_plan:
+    backup_plan_name: elastic
+    rules:
+    - RuleName: every_morning
+      TargetBackupVaultName: elastic
+      ScheduleExpression: "cron(0 5 ? * * *)"
+      StartWindowMinutes: 120
+      CompletionWindowMinutes: 10080
+      Lifecycle:
+        DeleteAfterDays: 7
+      EnableContinuousBackup: true
+  register: elastic_plan
+- name: show results
+  ansible.builtin.debug:
+    var: elastic_plan
+"""
+RETURN = """
+backup_plan_name:
+  description: backup plan name
+  returned: always
+  type: str
+  sample: elastic
+backup_plan_id:
```

Comment:
I donot see backup_plan_name and backup_plan_id as part of the response structure in the [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/backup/client/create_backup_plan.html). Are you adding these keys to the return dict?

Reply:
many thanks, very good catch, it really returned a `null` when creating a plan, fixed with 99b2dabc